server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java

```diff
@@ -114,4 +114,12 @@ public AggregatorFactory getParent() {
     public String getStatsSubtype() {
         return OTHER_SUBTYPE;
     }
+
+    protected boolean supportsConcurrentSegmentSearch() {
```

Comment:
nit: lets add java doc for the method something along the lines of:

```
Implementation should override this method and return true if the Aggregator created by the factory works with concurrent segment search execution model
```

Reply:
Added!